### DISCUSSION TITLE
Seperate gene tree files from .treefile

### DISCUSSION THREAD

**Discussion by hutumpyancha** [iqtree/iqtree2#254] [Q&A] [answered]
I have made my genetrees in iqtree and the output is a .treefile. The code I had used is - 
iqtree -S /work/upasanag/iqtree/gene-tree-50p/attempt3/locus-alignment-50p/ --prefix IQtree-gene-50p3 -T AUTO
I also have a .treefile for the concatenated alignment. Now I am trying to run SortaDate. It requires each gene tree to be an individual file with either .r.tree or .tre.rr extennsions (https://github.com/FePhyFoFum/SortaDate).Does anyone know how to transform all the gene trees in the .treefile to individual files with the above mentioned extensions? 
Thank you so much,
Upasana

**Comment by roblanf** [comment, marked as answer]
@hutumpyancha, this is not really a question about IQ-TREE, but see if this bash code works:

```bash
TREEFILE="/work/upasanag/iqtree/gene-tree-50p/attempt3/locus-alignment-50p/IQtree-gene-50p3.treefile"

OUTPUT_DIR="/work/upasanag/iqtree/gene-tree-50p/attempt3/locus-alignment-50p/individual_treefiles"

mkdir -p "${OUTPUT_DIR}"

# Read the .treefile and split into individual gene tree files
awk '{print $0 > sprintf("'"${OUTPUT_DIR}"'/gene_%d.r.tree", NR)}' "${TREEFILE}"
```

The last line should make files named to match the line in the input file. E.g. the 7th tree in the input file should appear in a file called `gene_7.r.tree`
